python/cuml/dask/linear_model/logistic_regression.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) 2019-2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from cuml.dask.common.base import BaseEstimator
+from dask_glm.estimators import LogisticRegression as LogisticRegressionGLM
+import cupy as cp
+import dask_cudf
+
+
+class LogisticRegression(BaseEstimator):
+
+    def __init__(self, client=None, verbose=False, **kwargs):
+        super(LogisticRegression, self).__init__(client=client,
+                                                 verbose=verbose,
+                                                 **kwargs)
+
+    def fit(self, X, y):
+        """
+        Fit the model with X and y.
+
+        Parameters
+        ----------
+        X : Dask cuDF dataframe or CuPy backed Dask Array (n_rows, n_features)
+            Features for regression
+        y : Dask cuDF Series or CuPy backed Dask Array (n_rows,)
+            Label (outcome values)
+        """
+        X = self._to_dask_cupy_array(X)
+        y = self._to_dask_cupy_array(y)
+        lr = LogisticRegressionGLM(**self.kwargs)
+        lr.fit(X, y)
+        self.lr = lr
+        return self
+
+    def predict(self, X):
+        """
+        Make predictions for X and returns a dask collection.
+
+        Parameters
+        ----------
+        X : Dask cuDF dataframe  or CuPy backed Dask Array (n_rows, n_features)
+            Distributed dense matrix (floats or doubles) of shape
+            (n_samples, n_features).
+
+        Returns
+        -------
+        y : Dask cuDF Series or CuPy backed Dask Array (n_rows,)
+        """
+        X = self._to_dask_cupy_array(X)
+        return self.lr.predict(X)
```

Comment:
Do you think we can use https://github.com/rapidsai/cuml/blob/5bb564e8f31b37809e7fe5880267582ea9e39481/python/cuml/dask/common/input_utils.py#L195 to return a type that matches the input given by the user?

Reply:
@dantegd it looks like these functions (to_output, to_dask_cupy, etc.) are designed to operate on futures rather than explicit collections like dask arrays or dataframes. If so, since this class is mostly a wrapper around execution in dask-glm, using a futures interface might require changing the upstream execution model which I think we want to avoid. Perhaps we could accomplish the same goal without relying on the futures interface by explicitly operating on the collection here, since this is in the "cuml.dask.extended" namespace?

Please correct me if I'm misunderstanding, as I'm not well versed in the cuml / dask interaction :)